<a href="https://colab.research.google.com/github/SagarSitaula1/Discrete_math/blob/main/project4_Counting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part 4 : CPSMA 3133 Dr. Jacob's Discrete Mathematics 

## Counting

### Sagar Sitaula and 267865




# Counting

In [9]:
possibleCard = ['hearts','spades','diamonds','clubs']

Since, I am using cards and there are four different types of symbol or shape to determine the card my posibilities in 10 sequences would be $4^{10}$.That means for every ten time I draw a card there will be four possibilities!

$$ 4*4*4\dots *4 = 4^{10} $$


In [10]:
4**10

1048576

If the order didnot matter our outcome might be different, So lets use a formula.If there are $n$ objects and $k$ slots but order does not matter, the number of ways to do this is

$$
\binom{n+k-1}{k-1} = \binom{n+k-1}{n}
$$

Since, there will be 10 objects it is just that there are only 4 categories that they can fall into.

In [11]:
import math

math.factorial(13)/(math.factorial(10)*math.factorial(3))

286.0

So, if the order is did not matter, there could be 286 numbe rof outcomes possible.

In [12]:
from itertools import combinations_with_replacement
print(list(combinations_with_replacement(possibleCard,10)))

[('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'spades'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'diamonds'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'clubs'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'spades', 'spades'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'spades', 'diamonds'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'spades', 'clubs'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'diamonds', 'diamonds'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'diamonds', 'clubs'), ('hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'hearts', 'he

In above I can see there are lot of element which kind of look like a right number of elements, but really hard to count.

In [13]:
len(list(combinations_with_replacement(possibleCard,10)))

286